pkg/volume/vsphere_volume/vsphere_volume.go

```diff
@@ -280,7 +281,9 @@ func (v *vsphereVolumeUnmounter) TearDownAt(dir string) error {
 		return fmt.Errorf("directory %s is not mounted", dir)
 	}
 
-	v.volPath = path.Base(refs[0])
+	//replacing the \040 encoding with space
```

Comment:
volumePath eg: "[local] xxx.vmdk" provided to attach/mount has a space between [local] and vmdk name and it is encoded as \040 when returned by GetMountRefs(). Due to that unmount fails to match the given volume path. So we need to explicitly replace it to make it work.